library/core/class.dispatcher.php

```diff
@@ -970,8 +966,10 @@ protected function makeCanonicalUrl($controller, \ReflectionFunctionAbstract $me
         // Add args after known names to the querystring.
         $query = [];
         $split = false;
+        $hasSender = false;
```

Comment:
What's the primary benefit of detecting something like this and adjusting `$args` in the callee versus making adjustments or indicating there's a sender from the caller?

Reply:
It was kind of a heuristic there. I was more worried that other endpoints could have an `$args` parameter and so I wanted to match the event handler as closely as possible.